x-pack/plugins/cases/server/connectors/cases/cases_connector_executor.ts

```diff
@@ -752,28 +746,44 @@ export class CasesConnectorExecutor {
     };
   }
 
-  private getCasesTitle(ruleName: string, oracleCounter: number) {
-    const suffix =
-      oracleCounter === INITIAL_ORACLE_RECORD_COUNTER
-        ? '(Auto-created)'
-        : `(${oracleCounter}) (Auto-created)`;
+  private getCasesTitle(
+    params: CasesConnectorRunParams,
+    grouping: GroupedAlerts['grouping'],
+    oracleCounter: number
+  ) {
+    const groupingDescription = Object.entries(grouping)
+      .map(([_, value]) => {
+        return convertValueToString(value);
+      })
+      .join(' & ');
 
-    const ruleNameTrimmed = ruleName.slice(0, MAX_TITLE_LENGTH - suffix.length - 1);
+    const suffix = `${
+      groupingDescription.length > 0 ? ` - Grouping by ${groupingDescription}` : ''
+    }${oracleCounter > INITIAL_ORACLE_RECORD_COUNTER ? ` (${oracleCounter})` : ''} (Auto-created)`;
```

Comment:
Should `(Auto-created)` be translated? What about the description or the grouping description?

Reply:
Hey! I do not think we should translate them because this content is generated dynamically on the server side. It is not part a static text on the UI.

Reply:
To be honest, I am not sure what is the best practice here and what we do in Kibana. @lcawl what we should do here?

Reply:
I'm not sure if either of these links are helpful: https://eui.elastic.co/#/utilities/i18n or https://github.com/elastic/kibana/tree/main/packages/kbn-i18n. However I agree that we should try to translate these phrases if it's possible.

Reply:
Done in [`81667dd` (#180758)](https://github.com/elastic/kibana/pull/180758/commits/81667dd9972ee314f666acb87262fc74d9a4deef)